core/contracts/financial_templates/implementation/PricelessPositionManager.sol

```diff
@@ -465,19 +466,22 @@ contract PricelessPositionManager is FeePayer {
         require(totalPositionCollateral.isGreaterThan(totalPaid));
         // TODO(#925): If this reverts here, then the position cannot expire. Collateral may be locked in contract.
 
-        // TODO(#873): add divCeil and mulCeil to make sure that all rounding favors the contract rather than the user.
-        // Adjust internal variables below.
         // Compute fee percentage that was paid by the entire contract (fees / collateral).
         // Unlike payFees, we are spreading fees across all locked collateral and NOT all PfC, which
         // implies that we are not forcing liquidations to be responsible for paying final fees when the position expires
-        FixedPoint.Unsigned memory feePercentage = totalPaid.div(totalPositionCollateral);
+
+        // TODO(#934. #925) we ceil() the fee percentage so that the contract behaves as if MORE fees were paid
+        // than in actuality. This causes the contract to act as if it has LESS collateral than in actuality.
+        // Therefore we avoid the situation in which there is MORE adjusted-collateral in the contract
+        // than in actuality. This could have the side effect of having leftover collateral in the contract, which we want to upper bound.
+        FixedPoint.Unsigned memory feePercentage = totalPaid.divCeil(totalPositionCollateral);
 
         // Compute adjustment to be applied to the position collateral (1 - feePercentage).
         FixedPoint.Unsigned memory adjustment = FixedPoint.fromUnscaledUint(1).sub(feePercentage);
 
         // Apply fee percentage to adjust totalPositionCollateral and positionFeeAdjustment.
-        totalPositionCollateral = totalPositionCollateral.mul(adjustment);
-        positionFeeAdjustment = positionFeeAdjustment.mul(adjustment);
+        totalPositionCollateral = totalPositionCollateral.mulCeil(adjustment);
+        positionFeeAdjustment = positionFeeAdjustment.mulCeil(adjustment);
```

Comment:
Same as above.

Reply:
same reply as above. In short, will revert back to `mul`